src/sync/sync_session.cpp

```diff
@@ -540,11 +540,16 @@ void SyncSession::handle_error(SyncError error)
             case ProtocolError::reuse_of_session_ident:
             case ProtocolError::bound_in_other_session:
             case ProtocolError::bad_message_order:
+            case ProtocolError::bad_client_version:
+            case ProtocolError::illegal_realm_path:
+            case ProtocolError::no_such_realm:
+            case ProtocolError::bad_changeset:
 #if REALM_SYNC_VER_MAJOR > 1
             case ProtocolError::bad_changeset_header_syntax:
             case ProtocolError::bad_changeset_size:
             case ProtocolError::bad_changesets:
             case ProtocolError::bad_decompression:
+            case ProtocolError::partial_sync_disabled:
```

Comment:
Is it really on purpose that all these errors are ignored?

Reply:
They are not being ignored, the session will still report the error. This switch just does special handling in some cases.